tests/e2e/test_isolation_level.py

```diff
@@ -0,0 +1,568 @@
+#!/usr/bin/env python3
+
+"""
+This script verifies whether the server is able to correctly
+detect and abort certain classes of transactional anomaly,
+inspired by the hermitage tests found at:
+    https://github.com/ept/hermitage
+and based on the correctness definitions laid out in:
+    https://pmg.csail.mit.edu/papers/adya-phd.pdf
+
+As of the time of committing this script, the database
+is able to meet the conditions required for Snapshot Isolation
+and Consistent View.
+
+All of these tests begin with a dataset of:
+    Kv { key: 1, value: 10 }
+    Kv { key: 2, value: 20 }
+
+and then assert various transactional correctness properties.
+"""
+
+import argparse
+import mgclient
+
+def setup(conn):
+    conn.autocommit = True
+    cursor = conn.cursor()
+    cursor.execute("CREATE CONSTRAINT ON (kv: Kv) ASSERT kv.key IS UNIQUE;")
+    conn.commit()
+    conn.autocommit = False
+
+    cursor = conn.cursor()
+    cursor.execute("MATCH (kv:Kv) DETACH DELETE kv;")
+    conn.commit()
+
+    update(cursor, 1, 10, "instantiate key 1")
+    conn.commit()
+    update(cursor, 2, 20, "instantiate key 2")
+    conn.commit()
+
+    assert_eq(get(cursor, 1), [10])
+    assert_eq(get(cursor, 2), [20])
+
+
+def update(cursor, key, value, comment = ""):
+    cursor.execute("""
+            // %s
+         MERGE (kv:Kv{ key: $key })
+           SET kv.value = $value
+        RETURN kv;
+    """ % comment, {'key': key, 'value': value})
+    ret = cursor.fetchall()
+    if len(ret) != 1:
+        print("expected ret to be len 1, but it's", ret)
```

Comment:
Favor f-strings formatting, it makes reading code more intuitive and it's a new optimized and more favoured  way to format a strings
```suggestion
        print(f"expected ret to be len 1, but it's {ret}")
```